pkg/workload/watcher.go

```diff
@@ -106,19 +94,13 @@ func NewWatcher(ctx context.Context, ns string, enabledWorkloadKinds []Kind) (Wa
 	return w, nil
 }
 
-func hasValidReplicasetOwner(wl k8sapi.Workload, enabledKinds []Kind) bool {
+func hasValidReplicasetOwner(wl k8sapi.Workload, enabledKinds k8sapi.Kinds) bool {
 	for _, ref := range wl.GetOwnerReferences() {
 		if ref.Controller != nil && *ref.Controller {
-			switch ref.Kind {
-			case "Deployment":
-				if slices.Contains(enabledKinds, DeploymentKind) {
-					return true
-				}
-
-			case "Rollout":
-				if slices.Contains(enabledKinds, RolloutKind) {
-					return true
-				}
+			kind := k8sapi.Kind(ref.Kind)
+			switch kind {
+			case k8sapi.DeploymentKind, k8sapi.RolloutKind:
+				return enabledKinds.Contains(kind)
```

Comment:
this introduces a tiny change in the behavior that might be unwanted.
Previously, if the kind wasn't listed as enabled, we'd continue the iteration as possibly there would be another kind that's enabled.

Reply:
There's no point in continuing the iteration. An object can only have one controller, and once the condition on line 111/99 is true, that controller has been found.